### PR TITLE
feat: Drill ODBC/JDBC Impersonation feature

### PR DESCRIPTION
**Pull request by Z0ltrix**
### SUMMARY

Added Drill ODBC/JDBC Impersonation feature and necessary translations/docs

### TESTING INSTRUCTIONS

Connect Superset to a Drill Cluster via ODBC/JDBC and activate Impersonation via UI.

### ADDITIONAL INFORMATION

- [x] Has associated issue: #17352
- [ ] Required feature flags:
- [x] Changes UI
- [ ] Includes DB Migration (follow approval process in [SIP-59](https://github.com/apache/superset/issues/13351))
  - [ ] Migration is atomic, supports rollback & is backwards-compatible
  - [ ] Confirm DB migration upgrade and downgrade tested
  - [ ] Runtime estimates and downtime expectations provided
- [x] Introduces new feature or API
- [ ] Removes existing feature or API

### CODE REVIEW

**Comment by villebro** [comment]
Thanks for the PR @Z0ltrix ! This LGTM, but you need to address some linting issues (lines too long etc, please check the `python-lint` check). Ping @cgivre if you have time to also review this as you're most familiar with Drill.

**Comment by cgivre** [comment]
@villebro On it!

File: superset/db_engine_specs/drill.py [hunk 1]
```diff
@@ -68,3 +68,12 @@ def convert_dttm(cls, target_type: str, dttm: datetime) -> Optional[str]:
     def adjust_database_uri(cls, uri: URL, selected_schema: Optional[str]) -> None:
         if selected_schema:
             uri.database = parse.quote(selected_schema, safe="")
+
+    @classmethod
+    def modify_url_for_impersonation(cls, url: URL, impersonate_user: bool, username: Optional[str]) -> None:
+        if impersonate_user and username is not None:
+            if url.drivername == "drill+odbc":
+                url.query["DelegationUID"] = username
```
Comment (eschutho) [inline superset/db_engine_specs/drill.py:76]: it looks like this query param applies to odbc and jdbc but not sadrill, is that correct?
cc @betodealmeida

Comment (Z0ltrix) [reply superset/db_engine_specs/drill.py:76]: For JDBC the parameter would be `impersonation_target` as you can see here --> https://drill.apache.org/docs/configuring-inbound-impersonation/ but we have not tested this.

Should i include a case for `drill+jdbc` with this parameter even if i have not tested this?

**Comment by Z0ltrix** [comment]
> Thanks for the PR @Z0ltrix ! This LGTM, but you need to address some linting issues (lines too long etc, please check the `python-lint` check). Ping @cgivre if you have time to also review this as you're most familiar with Drill.

Hi @villebro ,
code cleanup done... at least pycharm isnt complaining about this part.
Everything else?
Regards,
Christian

File: superset/db_engine_specs/drill.py (hunk 1 above)
Comment (villebro) [reply superset/db_engine_specs/drill.py:84]: @Z0ltrix I personally think you can include it if it is expected to work even if you haven't been able to test it yourself; if it doesn't someone else can continue the work by posting a follow-up PR.

**Review by villebro — COMMENTED** [review]
One comment regarding the jdbc case as pointed out by @eschutho . I'd still like to hear @cgivre 's thoughts, and a unit test to cover both odbc and jdbc cases would be super nice.

File: superset/db_engine_specs/drill.py (hunk 1 above)
Comment (Z0ltrix) [reply superset/db_engine_specs/drill.py:84]: added impersonation_target for jdbc

**Review by cgivre — APPROVED** [review]
LGTM +1  
Thanks for including the JDBC impersonation as well.

**Comment by villebro** [comment]
There appears to be some unrelated flakiness on CI. Let me look into what's going on..

**Comment by villebro** [comment]
@Z0ltrix can you rebase the PR? The flaky test has now been addressed, so this should now pass CI.

**Comment by Z0ltrix** [comment]
> @Z0ltrix can you rebase the PR? The flaky test has now been addressed, so this should now pass CI.

@villebro merged with current master and added some unittest.

**Review by villebro — APPROVED** [review]
LGTM, thanks for the unit tests; great template to add to other connectors that support impersonation! ❤️

**Comment by villebro** [comment]
@Z0ltrix there's one remaining linting error that needs to be resolved before we can hit merge

**Comment by Z0ltrix** [comment]
> @Z0ltrix there's one remaining linting error that needs to be resolved before we can hit merge

@villebro reformatted with black formatter, hope this helps

File: superset/db_engine_specs/drill.py (hunk 1 above)
Comment (Z0ltrix) [reply superset/db_engine_specs/drill.py:84]: @villebro , @eschutho ,
can we resolve this conversation?
Regards,
Christian

Comment (eschutho) [reply superset/db_engine_specs/drill.py:84]: Thanks @Z0ltrix and @villebro. Looks good!

**Pull request merged**